graphprot/prepare_graphprot_seqs.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env python
+
+# draft implementation
+# * TODO:
+#   * centering should be optional
+#   * viewpoint should be optional
+#   * check for nonunique ids and warn
+#   * check for bedtools version
+#   * write bed files for sequence coordinates
+#   * set rnd init for shuffling to have reproducible results
+#   * use my own temporary sequence files, properly clean up afterwards
+#   * check if seq length and core length arguments match or handle properly
+#   * handle input/output error gracefully
+#   * check if input bed coordinates are stranded
+
+from __future__ import print_function
+import argparse
+from csv import reader
+from itertools import izip
+import logging
+from pybedtools.featurefuncs import midpoint
+from pybedtools.helpers import get_chromsizes_from_ucsc
+from pybedtools import BedTool
+
+# parse command line arguments
+# positional arguments
+parser = argparse.ArgumentParser(
+    description="Create coordinates and fasta sequences for use with GraphProt.")
+parser.add_argument(
+    "bsites_fn", help="Path to binding site coordiantes in bed format")
+parser.add_argument(
+    "genome_id", help="Genome UCSC id")
+parser.add_argument(
+    "genome_fa_fn", help="Genome fasta sequences")
+# optional arguments
+parser.add_argument(
+    "--seq_length",
+    type=int,
+    default=150,
+    help="Length of sequences to create")
+parser.add_argument(
+    "--core_length",
+    type=int,
+    default=48,
+    help="Length of viewpoint region at center of sequence")
+parser.add_argument(
+    "--output_file_prefix",
+    default="",
+    help="Prefix to use for output filenames")
+parser.add_argument(
+    "--chromosome_limits",
+    help="Path to file containing chromosome limites as required by bedtools. Use this parameter disables automatic lookup via the genome id.")
+parser.add_argument(
+    "--negative_site_candidate_regions_fn",
+    help="Path to regions considered for placement of negatives in bed format")
+parser.add_argument(
+    "-v", "--verbose",
+    help="Be verbose.",
+    action="store_true")
+parser.add_argument(
+    "-d", "--debug",
```

Comment:
yes, as in notebooks errors are rendered in pink, while I use the logger levels just to vary the amount of info displayed